e2e/clone_test.go

```diff
@@ -0,0 +1,254 @@
+			stdout, stderr, err := kubectl("get", "-n", nsCloneTest, "pvc", volName, "-o", "json")
+			if err != nil {
+				return fmt.Errorf("failed to get PVC. stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+			}
+
+			var pvc corev1.PersistentVolumeClaim
+			err = json.Unmarshal(stdout, &pvc)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal PVC. stdout: %s, err: %v", stdout, err)
+			}
+			if pvc.Status.Phase != corev1.ClaimBound {
+				return fmt.Errorf("PVC %s is not bound", volName)
+			}
+			return nil
+		}).Should(Succeed())
+
+		By("creating clone of the PVC")
+		thinPVCCloneYAML := []byte(fmt.Sprintf(thinPvcCloneTemplateYAML, thinClonePVCName, "thinvol", pvcSize))
+		stdout, stderr, err = kubectlWithInput(thinPVCCloneYAML, "apply", "-n", nsCloneTest, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+
+		thinPodCloneYAML := []byte(fmt.Sprintf(thinPodCloneTemplateYAML, thinClonePodName, thinClonePVCName, nodeName))
+		stdout, stderr, err = kubectlWithInput(thinPodCloneYAML, "apply", "-n", nsCloneTest, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+
+		By("validating that the cloned volume is present")
+		Eventually(func() error {
+			stdout, stderr, err := kubectl("get", "-n", nsCloneTest, "pvc", thinClonePVCName, "-o", "json")
+			if err != nil {
+				return fmt.Errorf("failed to get PVC. stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+			}
+
+			var pvc corev1.PersistentVolumeClaim
+			err = json.Unmarshal(stdout, &pvc)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal PVC. stdout: %s, err: %v", stdout, err)
+			}
+			if pvc.Status.Phase != corev1.ClaimBound {
+				return fmt.Errorf("PVC %s is not bound", thinClonePVCName)
+			}
+			return nil
+		}).Should(Succeed())
+
+		Eventually(func() error {
+			volumeName, err = getVolumeNameofPVC(thinClonePVCName, nsCloneTest)
+			return err
+		}).Should(Succeed())
+
+		var lv *thinlvinfo
+		Eventually(func() error {
+			lv, err = getThinLVInfo(volumeName)
+			return err
+		}).Should(Succeed())
+
+		vgName := "node1-myvg4"
+		if isDaemonsetLvmdEnvSet() {
+			vgName = "node-myvg5"
+		}
+		Expect(vgName).Should(Equal(lv.vgName))
+
+		poolName := "pool0"
+		Expect(poolName).Should(Equal(lv.poolName))
+
+		By("deleting the source volume and application")
+		thinPodYAML = []byte(fmt.Sprintf(thinPodTemplateYAML, "thinpod", volName, nodeName))
+		stdout, stderr, err = kubectlWithInput(thinPodYAML, "delete", "-n", nsCloneTest, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+
+		thinPvcYAML = []byte(fmt.Sprintf(thinPVCTemplateYAML, volName, pvcSize))
+		stdout, stderr, err = kubectlWithInput(thinPvcYAML, "delete", "-n", nsCloneTest, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+
+		By("validating if the cloned volume is present and is not deleted")
+		Eventually(func() error {
```

Comment:
If I understand correctly,  the cloned volume must exist consistently rather than eventually.
Using `Consistenly()` or just checking once might be better.

Reply:
Done.